### PR TITLE
fix(emqx_gateway_api): don't crash on unknown status

### PR DESCRIPTION
**Pull request by sstrigler**

### CODE REVIEW

File: apps/emqx_gateway/src/emqx_gateway_api.erl [hunk 1]
```diff
@@ -71,12 +71,13 @@ paths() ->
...
 gateway(get, Request) ->
     Params = maps:get(query_string, Request, #{}),
-    Status =
-        case maps:get(<<"status">>, Params, undefined) of
-            undefined -> all;
-            S0 -> binary_to_existing_atom(S0, utf8)
-        end,
-    {200, emqx_gateway_http:gateways(Status)};
+    Status = maps:get(<<"status">>, Params, <<"all">>),
+    case lists:member(Status, [<<"all">>, <<"running">>, <<"stopped">>, <<"unloaded">>]) of
```
Comment (zmstone) [inline apps/emqx_gateway/src/emqx_gateway_api.erl:75]: we should investigate if swagger supports enum for query string schema.

https://github.com/emqx/emqx/pull/9225/files#diff-e5853e317e87d9c01cc7193b42a7f45f677ba85e5da0e29df9327fc474e635cfL243-L244

Comment (zmstone) [reply apps/emqx_gateway/src/emqx_gateway_api.erl:75]: maybe put the allowed list in `emqx_gateway_api_i18n.conf`

Comment (sstrigler) [reply apps/emqx_gateway/src/emqx_gateway_api.erl:75]: I'm not exactly happy hardcoding the list of accepted values here, but failed to come up with a better idea. It needs to match the spec header at https://github.com/emqx/emqx/blob/master/apps/emqx_gateway/src/emqx_gateway_http.erl#L97

Maybe someone has an idea?

Make it a macro (that's only used once?)? Place it where?
I haven't quite understood what `typerefl` can be used for, maybe there's a way?

Also I feel like the separation between `emqx_gateway_api` and `emqy_gateway_http` is a bit off. Can't pin-point it yet though.

Comment (sstrigler) [reply apps/emqx_gateway/src/emqx_gateway_api.erl:75]: > maybe document the allowed list in `emqx_gateway_api_i18n.conf`
> 
> edit: the scheme doc as defined in `?DESC(gateway_status_in_qs)` is currently missing from the i18n file.

Isn't it this here? https://github.com/sstrigler/emqx/blob/master/apps/emqx_gateway/i18n/emqx_gateway_api_i18n.conf#L67

Comment (zmstone) [reply apps/emqx_gateway/src/emqx_gateway_api.erl]: > > maybe document the allowed list in `emqx_gateway_api_i18n.conf`
> > edit: the scheme doc as defined in `?DESC(gateway_status_in_qs)` is currently missing from the i18n file.
> 
> Isn't it this here? https://github.com/sstrigler/emqx/blob/master/apps/emqx_gateway/i18n/emqx_gateway_api_i18n.conf#L67

Sorry, my code grep did not serve well.
Yes, you found it!

**Comment by HJianBo** [comment]
Sorry, I made a mistake while trying to fix Changes in Chinese. It caused `emqx/master` to be pushed to your master branch. 

So, this PR was automatically closed. **Sorry for the trouble.** :cry:  

I opened a new PR https://github.com/sstrigler/emqx/pull/1 to your master branch. When it is merged, the current PR may be able to re-open.

**Comment by sstrigler** [comment]
New PR here: https://github.com/emqx/emqx/pull/9258

**Pull request closed without merging**